src/languageservice/parser/jsonParser07.ts

```diff
@@ -1535,19 +1526,30 @@ function validate(
           validationResult: subValidationResult,
           matchingSchemas: subMatchingSchemas,
         };
-      } else if (compareResult === 0) {
+      } else if (
+        compareResult === 0 ||
+        ((node.value === null || node.type === 'null') && node.length === 0) // node with no value can match any schema potentially
```

Comment:
null yaml value is a good match for sub-schemas, similar to `isAutoCompleteEqualMaybe` for const and enums